microbial_thermodynamics/scipy_demo.py

```diff
@@ -0,0 +1,129 @@
+        y: An array of all changes of the three equations.
+
+    Returns:
+        The solution for all equations in the set
+    """
+
+    # Extract the relevant values from the vector of values y, then supply them to the
+    # relevant functions to calculate the change in each variable set
+
+    change_in_N = dN(
+        t,
+        N=y[0:number_of_species],
+        a=y[2 * number_of_species : 3 * number_of_species],
+        R=y[number_of_species : 2 * number_of_species],
+    )
+    change_in_r = dr(
+        t,
+        a=y[2 * number_of_species : 3 * number_of_species],
+        R=y[number_of_species : 2 * number_of_species],
+    )
+    change_in_a = da(
+        t,
+        a=y[2 * number_of_species : 3 * number_of_species],
+        R=y[number_of_species : 2 * number_of_species],
+    )
+
+    # Then combine these changes into a single vector and return that
+    return np.concatenate((change_in_N, change_in_r, change_in_a))
+
+
+def integrate() -> NDArray[np.float32]:
+    """Integrate the model using solve_ivp.
+
+    Returns:
+        These results of the integration.
+    """
+
+    # Set the update time (in seconds)
+    update_time = 100.0
+    # Use this to make the time span (starts at time = 0s)
+    t_span = (0.0, update_time)
+
+    # In this case we simulate 2 lots of the same set of equations
+    number_of_species = 2
+
+    # This means that each variable has 2 initial conditions
+    N0 = [23.0, 2.3]
+    r0 = [0.3, 0.3]
+    a0 = [1e6, 1e6]
+
+    # Construct vector of initial values y0
+    y0 = np.concatenate((N0, r0, a0))
+    # Carry out simulation, by supplying the set of equations, time span, initial
+    # condition, and any extra arguments
+    output = solve_ivp(
+        full_equation_set,
+        t_span,
+        y0,
+        args=(number_of_species,),
+    )
+    return output
+
+
+def run_and_plot_population() -> None:
+    """Runs the integrate function and plots population against time."""
+    plot_time = integrate().t
+    species_1_population = integrate().y[0]
+    species_2_population = integrate().y[1]
+    plt.plot(plot_time, species_1_population, label="species 1")
+    plt.plot(plot_time, species_2_population, label="species 2")
+    plt.xlabel("Time")
+    plt.ylabel("Population")
+    plt.title("Population change")
+    plt.show()
+
+
+def run_and_plot_r() -> None:
+    """Runs the integrate function and plots ribosome fraction against time."""
+    plot_time = integrate().t
+    species_1_r = integrate().y[2]
+    species_2_r = integrate().y[3]
+    plt.plot(plot_time, species_1_r, label="species 1")
+    plt.plot(plot_time, species_2_r, label="species 2")
+    plt.xlabel("Time")
+    plt.ylabel("Ribosome fraction")
+    plt.title("Ribosome fraction change")
+    plt.show()
+
+
+def run_and_plot_a() -> None:
+    """Runs the integrate function and plots internal energy (ATP) production against
+    time.
+    """  # noqa: D205
```

Comment:
Use the `# noqa` is smart here, but personally think it would be better to just shorten the docstring. Maybe use """Integrate set of equations and plot internal energy (ATP) against time.""" as the docstring instead